components/cards.css

```diff
@@ -0,0 +1,272 @@
+    display: flex;
+    gap: 2rem;
+    align-items: center;
+    flex-wrap: wrap;
+}
+
+.card{
+    font-size: 1rem;
+    width: fit-content;
+    height: fit-content;
+    box-shadow: 0 0 5px rgb(0 0 0 / 25%);
+}
+
+.basic-card{
+   
+    width: 15rem;
+    border-radius: 0.25rem;
+}
+.card-content{
+    padding: 0.5rem 1rem;
+}
+.card-header{
+    font-size: 1.125em;
+    font-weight: 600;
+    margin: 0.5rem 0rem;
+    padding: 0rem;
+}
+.card-title, .card-sub-header{
+    color: var(--color-darker-gray);
+    padding: 0rem;
+    margin: 0.75rem 0rem;
+}
+.card-description{
+    margin: 0.5rem 0rem;
+    font-size: 0.9em;
+    color: var(--color-dark-gray);
+}
+.card-btn{
+    margin: 0rem;
+    font-size: 0.8em;
+}
+
+/*---------------------------------------Image Card------------------------------------------------*/
+
+.image-card{
+    width: 20rem;
+    border-radius: 0.25rem;
+}
+.card-image-container{
+    padding: 0rem;
+    margin: 0rem;
+    width: 100%;
+    height: 12rem;
+    border-radius: inherit;
+}
+.card-img{
+    width: 100%;
+    height: 100%;
+    border-radius: inherit;
+}
+.card-button-container{
+    margin: 1rem 0rem;
+    display: flex;
+    gap: 1rem;
+}
+
+/*---------------------------------------Overlay Card------------------------------------------------*/
+
+.overlay-card{
+    margin-top: 1rem;
+    width: 15rem;
+    height: 18rem;
+    overflow: hidden;
+    border-radius: 0.25rem;
+    background-color:var(--color-light-blue);
+    text-align: center;
+    box-shadow: 0 0 5px rgba(0, 0, 0, 0.25);
+    position: relative; 
+}
+
+.overlay-card:hover{
+    box-shadow: 0 0 15px rgba(0, 0, 0, 0.25);
+}
+.overlay-card-image{
+    height: 100%;
+    width: 100%;
+    object-fit:fill;
+}
+.overlay-text{
+    position: absolute;
+    top: 0;
+    left: 0;
+    width: 100%;
+    height: 100%;
+    background: rgba(0, 0, 0, 0.6);
+    color: var(--color-white);
+    display: flex;
+    flex-direction: column;
+    flex-wrap: wrap;
+    align-items: center;
+    justify-content: center;
+    gap: 1rem;
+    opacity: 0;
+    transition: opacity 0.25s;
+}
+.overlay-text:hover{
+    opacity: 1;
+}
+.overlay-text > * {
+    transform: translateY(20px);
+    transition: transform 0.25s;
+}
+.overlay-text:hover > * {
+    transform: translateY(0);
+}
+.overlay-img-description{
+    font-size: 1.25rem;
+    font-weight: 600;
+}
+
+/*---------------------------------------Product Card------------------------------------------------*/
+
+.product-card{
+    width: 13rem;
+    border-radius: 0.25rem;
+    background-color:var(--color-light-blue);
+    text-align: center;
+    box-shadow: 0 0 7px rgba(0, 0, 0, 0.25);
+    display: flex;
+    flex-direction: column;
+    justify-content: space-between;
+    gap: 0rem;
+    position: relative; 
+}
+.product-card:hover{
+    box-shadow: 0 0 15px rgba(0, 0, 0, 0.25);
+}
+.product-image{
+    width: 100%;
+}
+.product-image img{
+    width: 100%;
+    object-fit: cover;
+}
```

Comment:
why mention the same property (width) again below?

Reply:
product-image is the container and img is its child, my bad I should have named it product-image-container.